### PR TITLE
build(components): buildCpuTrainer failure

### PR DESCRIPTION
**Pull request by jingzhang36**
**Description of your changes:**

We've observed persistent errors in buildCpuTrainer and buildGpuTrainer which seem to be caused by a version issue with setuptools. However, since we use tensorflow image 1.6.0, the latest setuptools via apt-get is 20.7.0. Therefore, in order to increase setuptools version, we increase tensorflow image to 2.3.0 and as a result, tensorflow transform and tensorflow model analysis need to be increased too. Let's see if this can get rid of the error.

Part of https://github.com/kubeflow/pipelines/issues/4361
Screen Shot 2020-08-24 at 5 18 39 PM

Screen Shot 2020-08-24 at 5 11 54 PM

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

File: components/kubeflow/dnntrainer/build_image.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ while getopts ":hp:t:i:b:l:" opt; do
         echo "-t: tag name"
         echo "-i: image name. If provided, project name and tag name are not necessary"
         echo "-b: tensorflow base image tag. Optional. The value can be tags listed under \
-        https://hub.docker.com/r/tensorflow/tensorflow/tags. Defaults to '1.6.0'."
+        https://hub.docker.com/r/tensorflow/tensorflow/tags. Defaults to '2.3.0'."
```
Comment (Bobgy) [inline components/kubeflow/dnntrainer/build_image.sh:22]: nit:
`0.23.0`?

Comment (Ark-kun) [reply components/kubeflow/dnntrainer/build_image.sh:22]: I think `2.3.0` is correct

File: components/kubeflow/dnntrainer/build_image.sh [hunk 2]
```diff
@@ -53,7 +53,7 @@ if [ -z "${TAG_NAME}" ]; then
 fi
 
 if [ -z "${TF_BASE_TAG}" ]; then
-  TF_BASE_TAG=1.6.0
+  TF_BASE_TAG=0.23.0
```
Comment (Ark-kun) [inline components/kubeflow/dnntrainer/build_image.sh:56]: `2.3.0`?

**Comment by Ark-kun** [comment]
Thank you for taking on this issue, Jing.

File: components/kubeflow/dnntrainer/build_image.sh (hunk 1 above)
Comment (jingzhang36) [reply components/kubeflow/dnntrainer/build_image.sh:22]: tensorflow is 0.23.0, while tensorflow transform and tensorflow model analysis is 2.3.0

File: components/kubeflow/dnntrainer/build_image.sh (hunk 2 above)
Comment (jingzhang36) [reply components/kubeflow/dnntrainer/build_image.sh]: Thank you for catching this. Revised.

**Comment by Ark-kun** [comment]
```
The following packages are not found for licenses tracking.
Please add an entry in /ml/third_party_licenses.csv for each of them.
argon2-cffi astunparse avro-python3 backcall importlib-metadata packaging QtPy tensorboard-plugin-wit zipp
```

**Pull request merged**